mrjones2014/legendary.nvim
# allow one-shot formatters

Resolves: #153 

## How to Test

1. Add the following keymap

```lua
vim.keymap.set({'n', 'v'}, '\\',
    function ()
        require('legendary').find({
            filters = { require('legendary.filters').current_mode() },
            formatter = function(item, mode)
                local values = require('legendary.formatter').get_default_format_values(item)
                if string.find(item.kind 'keymap') then
                    values[1] = mode
                end
                return values
            end
        })
    end
)
```

2. In each mode listed, press `\` and ensure that it filters keymaps by current mode, and that the first column renders only the current mode

## Testing for Regressions

I have tested the following:
